Cocos2d-x error: 'malloc' was not declared in this scope

https://github.com/elimu-ai/sima/actions/runs/14976040442/job/42068524500#step:11:826

```
[armeabi] Compile++ thumb: cocos2dx_internal_static <= CCUserDefault.cpp
[armeabi] Compile++ thumb: cocos2dx_internal_static <= CCValue.cpp
[armeabi] Compile++ thumb: cocos2dx_internal_static <= ObjectFactory.cpp
[armeabi] Compile++ thumb: cocos2dx_internal_static <= TGAlib.cpp
[armeabi] Compile++ thumb: cocos2dx_internal_static <= ZipUtils.cpp
[armeabi] Compile++ thumb: cocos2dx_internal_static <= CCAllocatorDiagnostics.cpp
In file included from /home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/./base/allocator/CCAllocatorGlobal.h:31:0,
                 from /home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/./base/allocator/CCAllocatorDiagnostics.cpp:28:
/home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/./base/allocator/CCAllocatorStrategyDefault.h: In member function 'void* cocos2d::allocator::AllocatorStrategyDefault::allocate(size_t)':
/home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/./base/allocator/CCAllocatorStrategyDefault.h:45:27: error: 'malloc' was not declared in this scope
         return malloc(size);
                           ^
/home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/./base/allocator/CCAllocatorStrategyDefault.h: In member function 'void cocos2d::allocator::AllocatorStrategyDefault::deallocate(void*, size_t)':
/home/runner/work/sima/sima/frameworks/cocos2d-x-3.10/cocos/./base/allocator/CCAllocatorStrategyDefault.h:51:25: error: 'free' was not declared in this scope
             free(address);
                         ^
At global scope:
cc1plus: warning: unrecognized command line option "-Wno-extern-c-compat"
cc1plus: warning: unrecognized command line option "-Wno-extern-c-compat"
make: *** [/home/runner/work/sima/sima/proj.android/obj/local/armeabi/objs/cocos2dx_internal_static/base/allocator/CCAllocatorDiagnostics.o] Error 1
make: Leaving directory `/home/runner/work/sima/sima/proj.android/jni'
Error: Process completed with exit code 2.
```

--- a/.github/workflows/android-build.yml
+++ b/.github/workflows/android-build.yml
@@ -79,10 +79,15 @@ jobs:
 
       - name: Patch Cocos2d-x files
         run: |
-          # Fix malloc/free not declared error by adding missing include to CCAllocatorStrategyDefault.h
-          file_path="$GITHUB_WORKSPACE/frameworks/cocos2d-x-3.10/cocos/base/allocator/CCAllocatorStrategyDefault.h"
-          sed -i '/#include "base\/allocator\/CCAllocatorMacros.h"/a #include <cstdlib>' "$file_path"
-          echo "Patched $file_path to include cstdlib"
+          # Fix malloc/free not declared error
+          file_path1="$GITHUB_WORKSPACE/frameworks/cocos2d-x-3.10/cocos/base/allocator/CCAllocatorStrategyDefault.h"
+          sed -i '/#include "base\/allocator\/CCAllocatorMacros.h"/a #include <cstdlib>' "$file_path1"
+          echo "Patched $file_path1 to include cstdlib"
+          
+          # Fix memset not declared error
+          file_path2="$GITHUB_WORKSPACE/frameworks/cocos2d-x-3.10/cocos/base/allocator/CCAllocatorStrategyGlobalSmallBlock.h"
+          sed -i '/#include "base\/allocator\/CCAllocatorMacros.h"/a #include <cstring>' "$file_path2"
+          echo "Patched $file_path2 to include cstring"
 
       - name: Debug dependency structure
         run: |

--- a/.github/workflows/android-build.yml
+++ b/.github/workflows/android-build.yml
@@ -77,6 +77,13 @@ jobs:
           
           echo "COCOS_X_ROOT=$GITHUB_WORKSPACE/frameworks" >> $GITHUB_ENV
 
+      - name: Patch Cocos2d-x files
+        run: |
+          # Fix malloc/free not declared error by adding missing include to CCAllocatorStrategyDefault.h
+          file_path="$GITHUB_WORKSPACE/frameworks/cocos2d-x-3.10/cocos/base/allocator/CCAllocatorStrategyDefault.h"
+          sed -i '/#include "base\/allocator\/CCAllocatorMacros.h"/a #include <cstdlib>' "$file_path"
+          echo "Patched $file_path to include cstdlib"
+
       - name: Debug dependency structure
         run: |
           cd $GITHUB_WORKSPACE/frameworks/cocos2d-x-3.10/external

--- a/.github/workflows/android-build.yml
+++ b/.github/workflows/android-build.yml
@@ -28,7 +28,7 @@ jobs:
 
       - name: Install Android SDK components
         run: |
-          sdkmanager "platforms;android-21" "build-tools;30.0.3"
+          sdkmanager "platforms;android-22" "build-tools;30.0.3"
 
       - name: Set up Android NDK r16b
         run: |

--- a/Kukua.udf
+++ b/Kukua.udf
@@ -14,7 +14,7 @@
         <LuaEncryptSign Value="" />
         <Android_PackageName Value="org.cocos.Kukua" />
         <AndroidkeyStore Value="" />
-        <AndroidVersion Value="android-21" />
+        <AndroidVersion Value="android-22" />
         <iOS_Target Value="Kukua-mobile" />
         <iOS_BundleID Value="iPhone Distribution: Melazeta srl" />
         <Platform Value="None" />

--- a/proj.android/AndroidManifest.xml
+++ b/proj.android/AndroidManifest.xml
@@ -5,7 +5,7 @@
     android:versionName="1.0"
     android:installLocation="auto">
 
-  <uses-sdk android:minSdkVersion="21" />
+  <uses-sdk android:minSdkVersion="22" />
 
   <uses-feature android:glEsVersion="0x00020000" />
 

--- a/proj.android/jni/Application.mk
+++ b/proj.android/jni/Application.mk
@@ -2,7 +2,7 @@ APP_STL := gnustl_static
 
 APP_CPPFLAGS := -frtti -DCC_ENABLE_CHIPMUNK_INTEGRATION=1 -std=c++11 -fsigned-char
 APP_LDFLAGS := -latomic
-APP_PLATFORM := android-21
+APP_PLATFORM := android-22
 
 
 ifeq ($(NDK_DEBUG),1)

--- a/proj.android/project.properties
+++ b/proj.android/project.properties
@@ -8,4 +8,4 @@
 # project structure.
 
 # Project target.
-target=android-21
+target=android-22

